src/data/reducers/assets.js

```diff
@@ -40,20 +41,17 @@ const status = (state = {}, action) => {
   }
 };
 
-const types = (state = {}, action) => {
+const parameters = (state = initialState, action) => {
   switch (action.type) {
     case assetActions.FILTER_UPDATED:
+      return { ...state, assetTypes: { ...state.assetTypes, ...action.data } };
```

Comment:
assetTypes is an object of key-value pairs where the key is filter name (e.g. OTHER, Images, etc.), and the value is whether or not the filter is selected (true or false, respectively). Since the assetTypes object should store all selected filters, we have to expand the current assetTypes and tack on the data of the action.